File: Data Structures & Algorithms/max-water-container/submission-1.go
func maxArea(heights []int) int {
    lp, rp := 0, len(heights) - 1
    ans := 0
    for lp < rp {
        ans = max(ans, min(heights[lp], heights[rp]) * (rp - lp))
        if heights[lp] < heights[rp] {
            lp++
        } else {
            rp--
        }
    }
    return ans
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        hm = {}
        ans = 0
        l, r, maxfreq = 0, 0, 0
        while r < len(s):
            if s[r] not in hm:
                hm[s[r]] = 0
            hm[s[r]] += 1
            # update the max occuring character count
            maxfreq = max(maxfreq, hm[s[r]])

            while (r - l + 1) - maxfreq > k:
                hm[s[l]] -= 1
                l += 1
            ans = max(ans, r - l + 1)
            r += 1
        return ans